Framework/Algorithms/src/PDFFourierTransform2.cpp

```diff
@@ -197,15 +196,16 @@ size_t PDFFourierTransform2::determineMaxIndex(double max, const std::vector<dou
     max = X.back();
   }
 
-  // get pointers for the data range
-  auto iter = std::lower_bound(X.begin(), X.end(), max);
+  // get index for the max from the X-range, tightening the range if a partial x bin value is provided
+  auto iter = std::upper_bound(X.begin(), X.end(), max) - 1;
   size_t max_index = std::distance(X.begin(), iter);
 
-  // go to first non-nan value
+  // go to first non-nan value. This works for histogram (bin edge) data and
+  // also point data, where point integration ends at the Y value to the left of the max bound
```

Comment:
Getting a bit pedantic now - but to be precise the integration for point data doesn't end at the Y value to the left of the max bound. It just uses the Y value at the left of each interval. So you do integrate the final interval that ends at max_index but the code just uses the Y value from the left hand edge of this interval